UnityProject/Assets/Scripts/Objects/Machines/Jukebox.cs

```diff
@@ -207,39 +191,40 @@ public async Task Stop(bool autoplay = false)
 			else
 				spriteHandler.SetSpriteSO(SpriteDamaged);
 
-			await Task.Run(() => WaitForActionToFinish(() => SoundManager.StopNetworked(guid)));
+			if(guid.Count != 0) await Task.Run(() => StopAllGuids());
 
 			UpdateGUI();
 		}
 
-		public void PreviousSong()
+		private Task StopAllGuids()
 		{
-			_ = Stop(true);
-			if (actionNotDone) return;
-			if (currentSongTrackIndex > 0)
+			foreach (var id in guid)
 			{
-				currentSongTrackIndex--;
-				UpdateGUI();
-
-				if (IsPlaying) _ = Play();
+				SoundManager.StopNetworked(id);
 			}
+			guid.Clear();
+
+			return Task.CompletedTask;
 		}
 
-		public bool NextSong()
+		public async void PreviousSong()
```

Comment:
this unity safe async?

Reply:
yes

Reply:
how is it referenced? as in not picking up the default c# version

Reply:
its not using anything related to unity so its fine

Reply:
unity only complains if you accessing the `gameObject` variable, also we've been using tasks forever now with jukeboxes and it works fine.

Reply:
hewwo??